src/cache.c

```diff
@@ -144,6 +145,36 @@ void CacheSystem_init(const char *path, int url_supplied)
     CACHE_SYSTEM_INIT = 1;
 }
 
+static int ntfw_cb(const char *fpath, const struct stat *sb, int typeflag, struct FTW *ftwbuf)
+{
+    (void) sb;
+    (void) typeflag;
+    (void) ftwbuf;
+    return remove(fpath);
+}
+
+void CacheSystem_clear(const char *path)
+{
+    char *cache_root_dir;
+    if (path) {
+        cache_root_dir = strdup(path);
+    } else {
+        char *xdg_cache_home = getenv("XDG_CACHE_HOME");
+        if (!xdg_cache_home) {
+            char *home = getenv("HOME");
```

Comment:
This is probably how the absence of HOME causes a crash in https://github.com/fangfufu/httpdirfs/issues/154

Reply:
You should refactor and re-use the existing instance of ``getenv("XDG_CACHE_HOME")``, rather than copying and pasting.

Reply:
I just want to clarify my reasoning. Considering this pull request is feature enhancement, and it is not difficult to delete the cache folder outside of this application, I would rather not add stuff that basically repeats #154. 

I would like you to factor out the extraction of ``XDG_CACHE_HOME`` , so both ``CacheSystem_calc_dir()`` and this share the same code path.